File: app/src/main/java/com/mine/dearbear/view/MainActivity.java
package com.mine.dearbear.view;

import android.app.Activity;
import android.os.Bundle;
import android.os.Handler;
import android.widget.Button;
import android.widget.Toast;

import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.viewholder.BaseViewHolder;
import com.dearbear.utilcode.socket.SocketServiceHelper;
import com.mine.dearbear.R;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;


public class MainActivity extends Activity {
    private String TAG = MainActivity.class.getName();
    private Button mTextBtn;
    private Handler mHandler = new Handler();
    private ItemAdapter mItemAdapter;
    private RecyclerView mRecyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initView();
        initData();
    }

    private void initView() {
        mRecyclerView = findViewById(R.id.mainRv);
    }

    private void initData() {
        List<String> data = new ArrayList<>();
        data.add("基本信息");
        data.add("应用信息");
        data.add("图片,音频");
        mItemAdapter = new ItemAdapter(data);
        mRecyclerView.setAdapter(mItemAdapter);
        mRecyclerView.setLayoutManager(new LinearLayoutManager(this));
//        mItemAdapter.notifyDataSetChanged();

//        Toast.makeText(this, "集成模式启动", Toast.LENGTH_LONG).show();
//        Intent intent = new Intent(this, ScrollingTestActivity.class);
//        startActivity(intent);
    }

    private void test() {
        SocketServiceHelper.getInstance().register(s -> {
            mHandler.post(() -> {
                Toast.makeText(this, s, Toast.LENGTH_SHORT).show();
            });
        });
    }


    static class ItemAdapter extends BaseQuickAdapter<String, BaseViewHolder> {

        public ItemAdapter(List<String> data) {
            super(R.layout.common_item_title_click, data);
        }

        @Override
        protected void convert(@NotNull BaseViewHolder baseViewHolder, String s) {
            baseViewHolder.setText(R.id.commonItemTitleTv, s);
        }
    }

}

File: app/src/main/java/com/mine/dearbear/view/MainActivity.kt
package com.mine.dearbear.view

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.widget.Button
import android.widget.Toast
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.viewholder.BaseViewHolder
import com.dearbear.utilcode.socket.SocketServiceHelper
import com.example.toolkit.ui.AppInfoActivity
import com.mine.dearbear.R
import com.mine.dearbear.databinding.ActivityMainBinding
import java.util.*

class MainActivity : Activity() {
    private val TAG = MainActivity::class.java.name
    private val mTextBtn: Button? = null
    private val mHandler = Handler()
    private var mItemAdapter: ItemAdapter? = null
    private var mRecyclerView: RecyclerView? = null
    var mBinding: ActivityMainBinding? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mBinding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(mBinding?.root)
        initView()
        initData()
    }

    private fun initView() {
        mRecyclerView = mBinding?.mainRv
    }

    private fun initData() {
        val data: MutableList<String> = ArrayList()
        data.add("基本信息")
        data.add("应用信息")
        data.add("图片,音频")

        mItemAdapter = ItemAdapter(data)
        mItemAdapter?.setOnItemClickListener { adapter, view, position ->
            if (position == 1) {
                val intent = Intent(this, AppInfoActivity::class.java)
                startActivity(intent)
            }
        }
        mRecyclerView?.adapter = mItemAdapter
        mRecyclerView?.layoutManager = LinearLayoutManager(this)
    }

    private fun test() {
        SocketServiceHelper.getInstance().register { s: String? ->
            mHandler.post {
                Toast.makeText(this, s, Toast.LENGTH_SHORT).show()
            }
        }
    }

    internal class ItemAdapter(data: MutableList<String>) : BaseQuickAdapter<String, BaseViewHolder>(R.layout.common_item_title_click, data) {
        override fun convert(baseViewHolder: BaseViewHolder, s: String) {
            baseViewHolder.setText(R.id.commonItemTitleTv, s)
        }
    }

}